### ISSUE TITLE
.gift seems like its not working properly (updated)

### ISSUE THREAD

**Issue by dorkix** [avatarkava/beavisbot-dubtrack-fm#112]
Hi, for some reason .gift @username is working properly but if someone tries to gift someone that has uppercase letters or numbers in the name it says "USER NOT FOUND" . Here's the error code from the console : 
<pre>
<code>
    Unhandled rejection TypeError: Cannot read property 'id' of undefined
    at /home/ubuntu/workspace/context.js:186:132
    at null.<anonymous> (/home/ubuntu/workspace/context.js:141:13)
    at tryCatcher (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/util.js:11:23)
    at Promise.module.exports.Promise._settlePromiseFromHandler (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/promise.js:489:31)
    at Promise.module.exports.Promise._settlePromise (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/promise.js:546:18)
    at Promise.module.exports.Promise._settlePromise0 (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/promise.js:591:10)
    at Promise.module.exports.Promise._settlePromises (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/promise.js:674:18)
    at Async._drainQueue (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/async.js:125:16)
    at Async._drainQueues (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/async.js:135:10)
    at Immediate.__dirname.drainQueues [as _onImmediate] (/home/ubuntu/workspace/node_modules/sequelize/node_modules/bluebird/js/release/async.js:16:14)
    at processImmediate [as _immediateCallback] (timers.js:374:17)

**Comment by avatarkava** [comment]
I was unaware capitalization was allowed on dubtrack - the code had been forcing every search to lowercase since usernames have always been forced into lowercase.  Has this changed?

One other way this could happen is if the user object in DubAPI is using isn't fully formed - they've been adding more functionality to that, so one thing you may want to try doing is running an `npm update` against the project to grab the latest copy of all the dependencies.  

Let me know how that goes and reopen if you're still running into the issue so we can diagnose and possibly open something on the DubAPI side or put in some local checks to protect against the error, since that's happening outside the Beavis code.

**Comment by dorkix** [comment]
Yes, Dubtrack now allows for uppercase usernames , also i ran  npm update :  
<pre>
<code>
manzorx:~/workspace (master) $ npm update
datatable@2.0.2 node_modules/datatable
└── underscore@1.4.4
cleverbot-node@0.2.0 node_modules/cleverbot-node
├── http@0.0.0
└── crypto@0.0.3
underscore@1.8.3 node_modules/underscore
npm WARN engine agentkeepalive@0.2.4: wanted: {"node":"0.10.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
dubapi@1.2.1 node_modules/dubapi
└── pubnub@3.7.18 (agentkeepalive@0.2.4)
<\pre>
<\code>
It is still happening , same exception stack.

**Comment by dorkix** [comment]
Update : changing var user = bot.getUserByName(usernameFormatted.toLowerCase()); to  var user = bot.getUserByName(usernameFormatted); seems to have fixed it.

**Comment by dorkix** [comment]
UPDATE : Dubtrack now allows for uppercase usernames and numbers.  Same exception stack, did npm update : 

<code>
manzorx:~/workspace (master) $ npm update
datatable@2.0.2 node_modules/datatable
└── underscore@1.4.4
cleverbot-node@0.2.0 node_modules/cleverbot-node
├── http@0.0.0
└── crypto@0.0.3
underscore@1.8.3 node_modules/underscore
npm WARN engine agentkeepalive@0.2.4: wanted: {"node":"0.10.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
dubapi@1.2.1 node_modules/dubapi
└── pubnub@3.7.18 (agentkeepalive@0.2.4)

**Comment by dorkix** [comment]
UPDATE : seems like changing : user = bot.getUserByName(usernameFormatted.toLowerCase());  to   var user = bot.getUserByName(usernameFormatted); fixed the problem.

**Comment by avatarkava** [comment]
That will fix it in part but will require that the user enters the name with *exactly* the same casing in order to find a match.  I'm going to open up a ticket with DubAPI to see if we can get that search to be case-insensitive on username so that we don't need to worry about entering everything exactly right.

**Comment by avatarkava** [comment]
Code updated on Beavis' end so now exact matching of case is the only thing supported.  Will wait on the Dubtrack response before I close this out.

**Comment by dorkix** [comment]
i dont wanna open another issue so ill post here, the update you did to config.json for controlling skipping stuck songs does not let bot start at all , throws some kind of error, if i delete skipStuckSongs from config.json it starts working. I updated bot.js already.

**Comment by avatarkava** [comment]
@dorkix if you can post the error it might help.  The update only adds to the sample config.json so it shouldn't have had any affect on your bot.  Additionally, nothing in config.json really could cause the bot to fail to start unless you put a syntax error in there - most common is forgetting the comma at the end of the line.

**Comment by dorkix** [comment]
Config.json is the same one from your updated one
<pre>
<code>
## Error
Debugger listening on port 15454
module.js:463
    throw err;
    ^

SyntaxError: /home/ubuntu/workspace/config.json: Unexpected token }
    at Object.parse (native)
    at Object.Module._extensions..json (module.js:460:27)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/ubuntu/workspace/bot.js:4:14)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
<pre/>
<code/>

## Config.json

  },
  "activeDJTimeoutMins": 60,
  "minActiveDJQueueLength": 5,
  "autoSuggestCorrections": false,
  "chatRandomnessPercentage": 1,
  "commandLiteral": ".",
  "customPointName": ":fire:",
  "developmentMode": false,
  "maxSongLengthSecs": 420,
  "prohibitDownvoteInQueue": false,
  "quietMode": false,
  "upvoteSongs": "ALL",
  "verboseLogging": false,
  "welcomeUsers": "ALL",
  "skipStuckSongs": true,
}

**Comment by avatarkava** [comment]
@dorkix No, it's not. Your last line in the block (skipStuckSongs) has a comma at the end, which is a syntax error for the last line of a block in JSON.

**Comment by dorkix** [comment]
Hey its me again, sorry for being annoying there's new problem now, after banning someone from the room, doing .idle crashes the bot with following error:

`at emitOne (events.js:77:13)
 at IncomingMessage.<anonymous> (/home/ubuntu/workspace/node_modules/dubapi/node_modules/pubnub/node.js/pubnub.js:2259:17)
 at doNTCallback2 (node.js:429:9)
 at process._tickCallback (node.js:343:17))`

and sometimes banning someone crashes the bot without doing any other command with following error 

`TypeError: Cannot read property 'username' of undefined
    at IncomingMessage.<anonymous> (/home/ubuntu/workspace/node_modules/dubapi/node_modules/pubnub/node.js/pubnub.js:2259:17)
    at doNTCallback2 (node.js:429:9)
    at process._tickCallback (node.js:343:17)`

**Comment by avatarkava** [comment]
In the future, please open a new issue for unrelated items, as I'll have to go and delete the comments again to try to keep this thread clean.

There's not enough of the stack trace there to properly diagnose the problem on the first issue.  The second doesn't include the full stack trace, either, but looks like a Dubtrack bug where it's not pushing the 'username' property on an incoming message.  That could be because the user left the room before the message was received.

**Comment by dorkix** [comment]
I created new issue sorry about that , you can delete my messages here. I apologize again.

**Comment by avatarkava** [comment]
Should be updated in latest version.  Make sure you run `npm update` to grab the latest dependencies.
